drivers/dma/dma_cavs.c

```diff
@@ -233,10 +233,18 @@ static int dw_dma_config(struct device *dev, u32_t channel,
 		cnt--;
 	} while (cfg_blocks && cnt);
 
-	/* Cyclic buffer not supported */
-	lli_desc_tail->llp = 0x0;
-	lli_desc_tail->ctrl_lo &=
-		~(DW_CTLL_LLP_S_EN | DW_CTLL_LLP_D_EN);
+	/* check if application requests circular list */
+	if (cfg_blocks) {
+		/*
+		 * if the last block was pointing to another block, then
+		 * it means the application is requesting a circular list
+		 */
+		lli_desc_tail->llp = (u32_t)chan_data->lli;
+	} else {
+		lli_desc_tail->llp = 0x0;
+		lli_desc_tail->ctrl_lo &=
+			~(DW_CTLL_LLP_S_EN | DW_CTLL_LLP_D_EN);
```

Comment:
How do we stop the dma in case of circular list? We need to somewhere execute the above statements for the dma to stop in case of circular list.

Reply:
stopping the dma will be done using the usual stop API. In case of dma_cavs, it will be through `dw_dma_transfer_stop`

Reply:
Yes. Theoretically this should work. But, this requires some stress testing (what if the DMA is in flight? things like that).

Reply:
The stop function is where the "graceful" stop needs to be addressed. I know currently the stop implementation is pretty simple. Since my use case is always listening, I am not focusing on the stop function. I'll address it in a separate PR.